### DISCUSSION TITLE
Incorrect filter/where clause in UPDATE query should raise error

### DISCUSSION THREAD

**Discussion by aalvrz** [sqlalchemy/sqlalchemy#9720] [Usage Questions]
I just experienced some very dangerous behavior with a very small mistake in a simple update query:

```python
query = Session.query(ModelA).filter(ModelB.some_id == some_val).update({"other_val": "foo"}, synchronize_session=False)
```

As you can see `ModelB` was mistakenly used in the filters instead of `ModelA`. SQLAlchemy does not complain and actually runs the query, ignoring the filter. This basically causes a full table update....

Why is this behavior allowed? Shouldn't SQLAlchemy raise an error if filters using a model that is actually not participating in the query are used?

---

SQLALchemy version: 1.4.46

**Comment by zzzeek** [comment]
an UPDATE can include WHERE criteria from other tables, your query outputs an UPDATE..FROM.

```py
>>> from sqlalchemy import table, column
>>> t1 = table('t1', column('a'))
>>> t2 = table('t2', column('b'))
>>> stmt = t1.update().where(t2.c.b == 'x').where(t2.c.b == t1.c.a)
>>> print(stmt)
UPDATE t1 SET a=:a FROM t2 WHERE t2.b = :b_1 AND t2.b = t1.a

```

**Reply by zzzeek** [reply]
we do have a "Cartesian product" warning that seems to not occur for UPDATE statements.  This can be implemented so will add an issue for it.

**Reply by zzzeek** [reply]
#9721.  should be in the next release
